doc/source/development/dev_practices.rst

```diff
@@ -16,6 +16,89 @@ of a RFC (request for comment) document.
 
 GDAL's policy on substantial code additions is documented at :ref:`rfc-85`.
 
+Portability
+-----------
+
+GDAL strives to be widely portable to 32 bit and 64 bit computing
+environments, as well as little-endian and big-endian ordered CPUs.
+CPL functions in the port directory provide services to abstract platform
+specific operations.
+
+Generally speaking, where available CPL functions should be used in
+preference to operating system functions for operations like memory
+allocation, path parsing, filesystem I/O (using VSILFILE* / VSIVirtualFile*),
+ODBC access, etc.
+
+C/C++ standards
+---------------
+
+The current C and C++ standards adopted by GDAL/OGR are C99 and C++17
+(last updated per :ref:`rfc-98`)`
+
+Variable naming
+---------------
+
+Much of the existing GDAL/OGR code uses an adapted Hungarian naming
+convention. Use of this convention is not mandatory, but when
+maintaining code using this convention it is desirable to continue
+adhering to it with changes. Most importantly, please avoiding using it
+improperly as that can be very confusing.
+
+In Hungarian prefixing the prefix tells something about about the type,
+and potentially semantics of a variable. The following are some prefixes
+used in GDAL/OGR.
+
+-  *a*: array
+-  *b*: C/C++ bool. In C code that pre-dates C99 adoption, it is also used for
+        ints with only TRUE/FALSE values.
+-  *by*: byte (GByte / unsigned char).
+-  *df*: floating point value (double precision)
+-  *e*: enumeration
+-  *i*: integer number used as a zero based array or loop index.
+-  *f*: floating point value (single precision)
+-  *h*: an opaque handle (such as GDALDatasetH).
+-  *n*: integer number (size unspecified)
+-  *o*: C++ object
+-  *os*: CPLString
+-  *p*: pointer
+-  *psz*: pointer to a zero terminated string. (eg. "char \*pszName;")
+-  *sz*: zero terminated string (eg." char szName[100];")
```

Comment:
```suggestion
-  *sz*: null-terminated string (eg." char szName[100];")
```

Reply:
should that be NUL-terminated string ? I believe NUL is the idimomatic way to designate character of code 0

Reply:
https://en.wikipedia.org/wiki/Null-terminated_string confirms your proposal. and also that a "null-terminate" string is terminated by character NUL. :-)